### PR TITLE
scan: Implement subscription hook for validators + wireup

### LINKED ISSUE #991: Implement subscription hook for validators + wireup
Both all validators and validator index. Mock data that's not available yet, but let's make sure ValidatorHook can be removed.

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #991

### CODE REVIEW

File: scan/.bsb 12.lock [hunk 1]
```diff
@@ -0,0 +1 @@
+84424
```
Comment (evilpeach) [inline scan/.bsb 12.lock]: remove

File: scan/src/subscriptions/ReportSub.re [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  };
+  // module MultiConfig = [%graphql
+  //   {|
+  //     subscription Reports($validator_address: String!, $limit: Int!, $offset: Int!) {
+  //       reports(where: {validator:{_eq: $validator_address}}, limit: $limit, offset:$offset) @bsRecord{
+  //         txHash: tx_hash @bsDecoder(fn: "Address.fromBech32")
+  //       }
+  //     }
+  //   |}
+  // ];
```
Comment (evilpeach) [inline scan/src/subscriptions/ReportSub.re]: remove?

File: scan/src/pages/ValidatorIndexPage.re [hunk 3]
```diff
@@ -77,124 +77,120 @@ let kvRow = (k, v: value_row_t) => {
...
 let make = (~address, ~hashtag: Route.validator_tab_t) =>
   {
-    let%Opt info = React.useContext(GlobalContext.context);
-    let%Opt validator =
-      info.validators
-      ->Belt_List.keep(({operatorAddress}) => operatorAddress->Address.isEqual(address))
-      ->Belt_List.get(0);
+    let validatorsSub = ValidatorSub.Validator.getList();
+    let validatorSub = ValidatorSub.Validator.get(address);
+    Js.Console.log(validatorSub);
```
Comment (evilpeach) [inline scan/src/pages/ValidatorIndexPage.re]: remove

File: scan/src/subscriptions/ValidatorSub.re [hunk 4]
```diff
@@ -1,7 +1,167 @@
...
+  | Unbonding;
+
 module Mini = {
   type t = {
     consensusAddress: string,
     operatorAddress: Address.t,
     moniker: string,
   };
 };
+module Validator = {
```
Comment (evilpeach) [inline scan/src/subscriptions/ValidatorSub.re]: I think we don't wrap in Validator.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/pages/ValidatorHomePage.re [hunk 5]
```diff
@@ -194,146 +194,117 @@ let renderBody = (rank, validator: ValidatorHook.Validator.t) => {
...
-  let validatorsOpt =
-    ValidatorHook.getList(
-      ~status=isActive ? ValidatorHook.Bonded : ValidatorHook.Unbonded,
-      ~page,
-      ~limit,
-      (),
-    );
+    let validatorsCountSub = ValidatorSub.count();
+    let validatorsSub = ValidatorSub.getList(~page, ~pageSize, ());
+    let bondedValidatorCountSub = ValidatorSub.count();
```
Comment (prin-r) [inline scan/src/pages/ValidatorHomePage.re:205]: add TODO

File: scan/src/pages/ValidatorIndexPage.re [hunk 6]
```diff
@@ -77,124 +77,119 @@ let kvRow = (k, v: value_row_t) => {
...
-      info.validators
-      ->Belt_List.keep(({operatorAddress}) => operatorAddress->Address.isEqual(address))
-      ->Belt_List.get(0);
+    let validatorsSub = ValidatorSub.getList();
+    let validatorSub = ValidatorSub.get(address);
+
+    let%Sub validators = validatorsSub;
+    let%Sub validator = validatorSub;
+
     let totalPower =
```
Comment (prin-r) [inline scan/src/pages/ValidatorIndexPage.re]: Create another Sub that return aggregation of voting power

Comment (prin-r) [reply scan/src/pages/ValidatorIndexPage.re]: We can't do that right now, so please add TODO.

File: scan/src/subscriptions/ValidatorSub.re [hunk 7]
```diff
@@ -1,7 +1,165 @@
...
+  nodeStatus: node_status_t,
+  delegators: list(delegator_t),
+};
+
+let toExternal = ({operatorAddress, moniker, identity, website}: internal_t) => {
+  avgResponseTime: 2,
+  isActive: true,
+  operatorAddress,
+  consensusPubKey:
+    "bandvalconspub1addwnpepq0grwz83v8g4s06fusnq5s4jkzxnhgvx67qr5g7v8tx39ur5m8tk7rg2nxj"
```
Comment (prin-r) [inline scan/src/subscriptions/ValidatorSub.re:57]: We are not using it right now, please remove this field.

**Review by prin-r — APPROVED** [review]

**Review by prin-r — CHANGES_REQUESTED** [review]
fix things first

**Review by prin-r — DISMISSED** [review]

File: scan/src/components/validator/ReportsTable.re [hunk 8]
```diff
@@ -22,222 +22,201 @@ module Styles = {
...
-                   spacing={Text.Em(0.05)}
-                 />
-               </Col>
-               <Col size=1.5>
-                 <div className={Styles.vFlex(`flexEnd)}>
-                   <div className=Styles.fillLeft />
+    let reportsSub =
+      ReportSub.ValidatorReport.getListByValidator(
+        ~page,
+        ~pageSize=5,
```
Comment (evilpeach) [inline scan/src/components/validator/ReportsTable.re]: use pageSIze

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by prin-r — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**